File: api_bus/core/models/ruta.py
from django.db import models
from .autobus import Autobus

class Ruta(models.Model):
    nombre = models.CharField(max_length=100)
    punto_inicio = models.CharField(max_length=100)
    punto_final = models.CharField(max_length=100)
    distancia_km = models.DecimalField(max_digits=6, decimal_places=2)
    tiempo_estimado = models.DurationField()
    autobus_asignado = models.ForeignKey(Autobus, on_delete=models.SET_NULL, null=True, blank=True)
    activo = models.BooleanField(default=True)
    
    def __str__(self):
        return f"{self.nombre} ({self.punto_inicio} - {self.punto_final})"

File: api-v2/models/ruta.js
import { DataTypes } from 'sequelize';
import sequelize from '../config/db.js';

const Ruta = sequelize.define(
  'Ruta',
  {
    // en cada modelo:
id: {
  type: DataTypes.BIGINT.UNSIGNED,
  autoIncrement: true,
  primaryKey: true,
},

    nombre: { type: DataTypes.TEXT, allowNull: true },
    punto_inicio: { type: DataTypes.TEXT, allowNull: true },
    punto_final: { type: DataTypes.TEXT, allowNull: true },
    distancia_km: { type: DataTypes.DECIMAL(10, 2), allowNull: true },
    // Guardamos duración en segundos
    tiempo_estimado_seg: { type: DataTypes.INTEGER.UNSIGNED, allowNull: true },

    autobus_asignadoId: {
      type: DataTypes.BIGINT.UNSIGNED,
      allowNull: true,
      references: { model: 'Autobuses', key: 'id' },
      onUpdate: 'CASCADE',
      onDelete: 'SET NULL',
    },

    activo: { type: DataTypes.BOOLEAN, allowNull: true, defaultValue: true },
  },
  {
    tableName: 'Rutas',
    timestamps: true,
    underscored: true,
  }
);

Ruta.associate = (models) => {
  Ruta.belongsTo(models.Autobus, {
    as: 'autobus_asignado',
    foreignKey: 'autobus_asignadoId',
  });
  Ruta.hasMany(models.Horario, { as: 'horarios', foreignKey: 'rutaId' });
};

export default Ruta;
